src/fs/rock/RockRebuild.cc

```diff
@@ -694,19 +697,41 @@ Rock::Rebuild::swanSong()
     storeRebuildComplete(&counts);
 }
 
+void
+Rock::Rebuild::callException(const std::exception &e)
+{
+    ErrorDetail::Pointer errorDetail;
+    if (const auto tex = dynamic_cast<const TextException*>(&e))
+        errorDetail = new ExceptionErrorDetail(tex->id());
+    else
+        errorDetail = new ExceptionErrorDetail(Here().id());
+    failure(ToSBuf(e.what(),
+        Debug::Extra, errorDetail).c_str());
+    // fatal failure() substitutes default AsyncJob::callException(e) handling
+}
+
 void
 Rock::Rebuild::failure(const char *msg, int errNo)
```

Comment:
In the method declaration, please make errNo a required argument (by removing the zero default). It may still be zero, but we expect the caller to supply `errno`.